src/index.js

```diff
@@ -1,47 +1,72 @@
-    completed: false,
-    index: 2,
-  },
-];
-tasks.sort((a, b) => a.index - b.index);
-function populateTodoList() {
-  const todoList = document.getElementById('todo-list');
-
-  for (let i = 0; i < tasks.length; i += 1) {
-    const { index, description, completed } = tasks[i];
-    todoList.innerHTML += `
-        
-       
-        <li id="L${index}" class ="common">
-        <div class = "list_container">
-        <div class = "list2">
-        <input for ="P${index}" id="${index}" type="checkbox" ${
+const clearButton = document.querySelector('.btnComplete');
+
+const input = document.getElementById('input');
+
+const target = document.getElementById('list');
+
+let tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+function saveToLocalStorage(data) {
+  tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  tasks.push(data);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+input.addEventListener('keyup', (event) => {
+  if (event.key === 'Enter') {
+    const inputValue = input.value;
+    const newObj = {
+      description: inputValue,
+      completed: false,
+      index: tasks.length + 1,
+    };
+    saveToLocalStorage(newObj);
+    window.location.reload();
+    input.value = '';
+  }
+});
+
+for (let i = 0; i < tasks.length; i += 1) {
+  const { index, description, completed } = tasks[i];
+  target.innerHTML += `
+      <li id="L${index}" class ="common">
+      <input for ="P${index}" id="${index}" type="checkbox" ${
   completed && 'checked'
 }  class ="checkbox">
-        <p id ="P${index}" class="li-p">${description}</p>
-        </div>
-        <button id="edit-remove${index}"  class="btn dots list-item">
-         <i class="fa fa-ellipsis-v"></i>
-        </button>
-        </li>
-        </div>
-      `;
-  }
+      <p id ="P${index}" class="li-p">${description}</p>
+      <button id="edit-remove${index}"  class="btn dots list-item">
+       <i class="fa fa-ellipsis-v"></i>
+      </button>
+      </li>
+    `;
 }
 
-// Call the function to populate the list on page load
-populateTodoList();
+const deleteBtn = document.querySelectorAll('.list-item');
+deleteBtn.forEach((button) => {
+  button.addEventListener('click', addTrash);
+});
+
+const checkboxs = document.querySelectorAll('.checkbox');
+checkboxs.forEach((el) => {
+  el.addEventListener('click', () => {
+    const list = JSON.parse(localStorage.getItem('tasks'));
+    const findIndex = list.findIndex(
+      (listEl) => listEl.index.toString() === el.id.toString(),
+    );
+
+    list[findIndex].completed = !list[findIndex].completed;
+    localStorage.setItem('tasks', JSON.stringify(list));
+    tasks = JSON.parse(localStorage.getItem('tasks'));
+  });
+});
+
+clearButton.addEventListener('click', () => {
+  const list = JSON.parse(localStorage.getItem('tasks'));
+  const filteredItems = list.filter((el) => el.completed === false);
+
+  localStorage.setItem('tasks', JSON.stringify(filteredItems));
+  window.location.reload();
```

Comment:
- [ ] Please 🙏🏻 remove the ```window.location.reload();``` function from your codebase because whenever a person tries to add or remove the tasks in your `TODO` app, then your page refreshes on every operation, which is actually not a good idea and not professional. Just assume you made an excellent website with some more functionalities and heavy design, and on every user operation if your page refreshes again and again, it can cause **BAD USER EXPERIENCE**, which can divert your page visitor and make them frustrated to wait for page first load properly before doing any operation or visit to another navigation menu.

[Screencast from 14-07-23 02:31:44 PM IST.webm](https://github.com/Zel-hub7/ToDoList/assets/66476329/f135a6c6-8c40-4017-9d80-eb65f8d58bf4)